gpu_tests/test_hub_utils.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+from metaseq.scripts.convert_to_singleton import create_generation_config_with_defaults
+from metaseq.hub_utils import GeneratorInterface, GeneratorHubInterface
+from metaseq.distributed import utils as distributed_utils
+from metaseq.dataclass.configs import MetaseqConfig
+import numpy as np
+from megatron.mpu import destroy_model_parallel
+from metaseq.distributed import fsdp_wrap, fsdp_enable_wrap
+from metaseq import hub_utils
+import torch.distributed as dist
+import torch
+
+
+PROMPT = [133, 313, 1224, 15, 5, 856, 17527, 594, 98, 5, 11471, 3820, 19, 514, 4]
+
+
+def default_post_build_model_hook(model, task):
+    # For fp16
+    model = model.half()
+    # fsdp_wrap will be a no-op if not using FSDP
+    if hasattr(model, "make_generation_fast_"):
+        model.make_generation_fast_()
+    return fsdp_wrap(model)
+
+
+def override_task(model):
+    # HACK: Use special version of the language_modeling task for inference with the models
+    # trained with streaming_language_modeling.
+    # TODO: More elegant solution would be to decouple the training and inference logic for the tasks
```

Comment:
I think we should be able to "language_modeling_inference_for_models_trained_with_streaming', the API only needs "language_modeling" to run.